pkg/blockbuilder/blockbuilder.go

```diff
@@ -441,17 +456,35 @@ func getGroupLag(ctx context.Context, admClient *kadm.Client, topic, group strin
 		return nil, err
 	}
 
-	// For now, if the group doesn't have any commits, we use startOffsets as the starting point.
-	if len(offsets) == 0 {
-		offsets = make(kadm.OffsetResponses)
-		for t, pt := range startOffsets.Offsets() {
-			resp := make(map[int32]kadm.OffsetResponse)
-			for _, o := range pt {
-				resp[o.Partition] = kadm.OffsetResponse{
-					Offset: o,
-				}
+	listOffsetsAfterMilli := sync.OnceValues(func() (kadm.ListedOffsets, error) {
+		return admClient.ListOffsetsAfterMilli(ctx, fallbackOffset, topic)
```

Comment:
How does this work when fallbackOffset is negative?

Reply:
Below there is a check

```
// If fallback is not a timestamp, it's the kafkaOffsetStart value.
if fallbackOffset < 0 {
    ...
```

I will make this chunk more explicit